core/src/main/scala/filodb.core/memstore/OnDemandPagingShard.scala

```diff
@@ -118,7 +120,9 @@ TimeSeriesShard(dataset, storeConfig, shardNum, bufferMemoryManager, rawStore, m
             Observable.fromIterable(partKeyBytesToPage.zip(pagingMethods))
               .mapAsync(storeConfig.demandPagingParallelism) { case (partBytes, method) =>
                 rawStore.readRawPartitions(dataset, allDataCols, SinglePartitionScan(partBytes, shardNum), method)
-                  .mapAsync { rawPart => partitionMaker.populateRawChunks(rawPart).executeOn(singleThreadPool) }
+                  .mapAsync { rawPart =>
+                    partitionMaker.populateRawChunks(rawPart).executeOn(singleThreadPool).asyncBoundary
```

Comment:
Actually, now that we have locks, I'm not sure this needs to be run on the ingestion thread anymore....  I think this was designed before the lock stuff was really in place.... Brian should comment... @broneill

Reply:
@velvia It is needed. The locks take care of chunk-map operations.
Partition creation modifies the shard state, namely PartitionSet, PartitionMap etc which need to be altered by ingestion thread only (unless we add additional locking).

Reply:
@vishramachandran  populateRawChunks does not modify `PartitionSet` etc. or any ingestion thread data structures.  The code only modifies ChunkMap, it's function is solely to copy the chunks to off heap.

Reply:
ok removing executeOn